internal/provisioner/cluster.go

```diff
@@ -30,15 +30,20 @@ func CreateCluster(provider, size string, logger log.FieldLogger) error {
 		return fmt.Errorf("unsupported size %s", size)
 	}
 
-	outputDir := "tmp"
-	_, err := os.Stat("tmp")
+	if clusterId == "" {
+		clusterId = model.NewId()
+	}
+
+	// Temporarily relocate the kops output directory to a local folder based on the
+	// cluster name. This won't be necessary once we persist the output to S3 instead.
+	outputDir := fmt.Sprintf("cluster-%s", clusterId)
+	_, err := os.Stat(outputDir)
 	if err == nil {
-		return errors.New("tmp folder already exists: delete existing cluster first")
+		return fmt.Errorf("encountered cluster ID collision: directory %q already exists", outputDir)
```

Comment:
`fmt.Errorf` is fine, but I've literally just discovered that one can also use `errors.Errorf`. That also gives us the extra stack information. Might be good to update everywhere.